### PR TITLE
test: cover registry persistence and preprocessor edges

### PR DESCRIPTION
**Pull request by Peyton-Spencer**
## Summary

- Add deterministic disk persistence tests for `UserRegistryService.load()` and `save()` with backup/restore of `/workspace/ipc/user_registry.json`.
- Add task preprocessor edge-case tests for empty stdout, default skip/error messages, whitespace scripts, and default workflow-dir traversal handling.
- Fix `runTaskPreprocessor()` so default workflow directory resolution errors return `{ action: 'error' }` instead of throwing before the guarded path setup.

## Test Count / Coverage

- Before: `bun test` on initial scheduled branch: 2238 tests, 6012 assertions; after switching to current `origin/main`: `bun test --coverage` reported 2358 tests, 6310 assertions, 91.19% line coverage.
- After: `bun test --coverage` reports 2365 tests, 6322 assertions, 91.56% line coverage.
- `src/effect/user-registry.ts`: 69.44% -> 100.00% line coverage.
- `src/task-preprocessor.ts`: 96.41% -> 99.21% line coverage.

## Verification

- `bun test src/task-preprocessor.test.ts src/effect/user-registry.test.ts`
- `bun run typecheck`
- `bunx prettier --check src/task-preprocessor.ts src/task-preprocessor.test.ts src/effect/user-registry.test.ts`
- `bun test`
- `bun test --coverage`

## Remaining Gaps

- Large integration-heavy modules remain the main uncovered areas: channel adapters, `src/index.ts`, `src/backends/local-backend.ts`, `src/discovery/routes.ts`, `src/db.ts`, and `src/web/server.ts`.

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for user registry disk persistence functionality.
  * Expanded test coverage for task preprocessor edge cases, including workflow output handling and path validation.

* **Bug Fixes**
  * Improved error handling in task preprocessing to ensure exceptions are properly caught and reported.

### CODE REVIEW

File: src/effect/user-registry.test.ts [hunk 1]
```diff
@@ -40,6 +42,26 @@ const charlie: UserInfo = {
...
+const registryPath = '/workspace/ipc/user_registry.json';
+let originalRegistryFile: string | null = null;
+let originalRegistryExisted = false;
+
+function backupRegistryFile() {
+  originalRegistryExisted = existsSync(registryPath);
+  originalRegistryFile = originalRegistryExisted
+    ? readFileSync(registryPath, 'utf-8')
+    : null;
+  mkdirSync(dirname(registryPath), { recursive: true });
```
Comment (Peyton-Spencer) [inline src/effect/user-registry.test.ts]: Blocking: this setup writes to the hard-coded runtime path `/workspace/ipc/user_registry.json`, and CI fails before the test body because the runner cannot create `/workspace` (`EACCES: permission denied, mkdir '/workspace'`). Even on machines where `/workspace` exists, this backup/restore pattern can clobber live IPC registry updates made during the test. Please make the registry path injectable for tests, or derive it from a temp IPC root/env override, and point these load/save cases at an `mkdtemp` directory instead of backing up the live path.

**Review by Peyton-Spencer — COMMENTED** [review]
Found one blocking issue in the requested disk persistence coverage.

Verified:
- `bun test src/task-preprocessor.test.ts` passes locally (20/20)
- `bun run typecheck` passes locally
- GitHub CI `test` fails on the new user-registry load/save tests with `EACCES: permission denied, mkdir '/workspace'`

The task preprocessor traversal handling looks correct; the blocker is the registry test setup writing to the hard-coded runtime IPC path.

File: src/effect/user-registry.test.ts [hunk 2]
```diff
@@ -40,6 +42,26 @@ const charlie: UserInfo = {
...
+}
```
[the remark this answers is not in the document]
Comment (Peyton-Spencer) [reply src/effect/user-registry.test.ts]: @/workspace/group/pr864-thread-reply.md

File: src/effect/user-registry.test.ts (hunk 1 above)
Comment (Peyton-Spencer) [reply src/effect/user-registry.test.ts]: @/workspace/group/pr864-thread-reply.md

**Review by Peyton-Spencer — COMMENTED** [review]
Rebased onto current `main` and force-pushed `0f7fcee`.

Resolution notes:
- Dropped the overlapping `user-registry` persistence test changes because newer temp-path coverage already exists on `main`.
- Kept the task-preprocessor fix/tests: omitted skip/error details, empty stdout fallback, whitespace script rejection, and group-folder traversal error handling.

Verified locally:
- `bun test src/task-preprocessor.test.ts src/effect/user-registry.test.ts` — 43 pass / 0 fail
- `bun run typecheck` — pass

Current GitHub state after push: mergeable, CodeRabbit green, CI still running. Hold merge until the new CI run finishes green.

**Comment by Peyton-Spencer** [comment]
Follow-up: the post-rebase CI run is now green.

Current state: `CLEAN` / mergeable, all required checks pass, CodeRabbit green. No remaining blocking findings from me.
